Objects/longobject.c

```diff
@@ -4309,18 +4315,34 @@ long_rshift(PyLongObject *a, PyLongObject *b)
     }
     else {
         shiftby = PyLong_AsSsize_t((PyObject *)b);
-        if (shiftby == -1L && PyErr_Occurred())
-            return NULL;
-        if (shiftby < 0) {
-            PyErr_SetString(PyExc_ValueError,
-                            "negative shift count");
-            return NULL;
+        if (shiftby >= 0) {
+            wordshift = shiftby / PyLong_SHIFT;
+            loshift = shiftby - wordshift * PyLong_SHIFT;
```

Comment:
> Is it worth to extract it in the helper function?

That sounds reasonable to me, if you want to do it; if you don't, that also sounds reasonable to me. :-)